src/main/java/de/tum/in/www1/artemis/web/rest/ModelingExerciseResource.java

```
@@ -437,6 +437,11 @@ public ResponseEntity<ModelingPlagiarismResult> checkPlagiarism(@PathVariable lo
         long start = System.nanoTime();
         log.info("Start modelingPlagiarismDetectionService.checkPlagiarism for exercise {}", exerciseId);
         ModelingPlagiarismResult result = modelingPlagiarismDetectionService.checkPlagiarism(modelingExercise, similarityThreshold / 100, minimumSize, minimumScore);
+
+        if (result == null) {
+            throw new BadRequestAlertException("Only one active plagiarism check per course allowed", "PlagiarismCheck", "oneActivePlagiarismCheck");
```

please throw this exception directly in the `checkPlagiarism` method